psi4/src/psi4/dfocc/ccd_F_intr.cc

```diff
@@ -36,22 +36,130 @@ namespace psi {
 namespace dfoccwave {
 
 void DFOCC::ccd_F_intr() {
-    // defs
-    SharedTensor2d K, T, U, Tau;
-
-    // read
-    Tau = SharedTensor2d(new Tensor2d("T2 (Q|IA)", nQ, naoccA, navirA));
-    Tau->read(psio_, PSIF_DFOCC_AMPS);
-
-    // OO block
-    // F_mi +=  \sum_{Q,e} Tau"_ie^Q b_me^Q
-    FijA->zero();
-    FijA->contract332(false, true, navirA, bQiaA, Tau, 1.0, 1.0);
-
-    // VV block
-    // F_ae -=  \sum_{Q,m} Tau'_ma^Q b_me^Q
-    FabA->contract(true, false, navirA, navirA, nQ * naoccA, Tau, bQiaA, -1.0, 0.0);
-    Tau.reset();
+
+    // RHF
+    if (reference_ == "RESTRICTED") {
+        // defs
+        SharedTensor2d K, T, U, Tau;
+
+        // OO block
+        // F_mi = (1-\delta_{mi}) f_mi
+        FijA->zero();
+        for(int m=0; m<naoccA; ++m) {
+             for(int i=0; i<naoccA; ++i) {
+                 int mi = (m * naoccA) + i;
+                 double value = 0.0;
+                 if (m != i) value = FockA->get(m+nfrzc, i+nfrzc);
+                 FijA->set(m, i, value);
+             }
+        }
+
+
+        // read
+        Tau = std::make_shared<Tensor2d>("T2 (Q|IA)", nQ, naoccA, navirA);
+        Tau->read(psio_, PSIF_DFOCC_AMPS);
+
+        // F_mi +=  \sum_{Q,e} Tau"_ie^Q b_me^Q
+        FijA->contract332(false, true, navirA, bQiaA, Tau, 1.0, 1.0);
+
+        // VV block
+        // F_ae = (1-\delta_{ae}) f_ae
+        FabA->zero();
+        for(int a=0; a<navirA; ++a) {
+             for(int e=0; e<navirA; ++e) {
+                 int ae = (a * navirA) + e;
+                 double value = 0.0;
+                 if (a != e) value = FockA->get(a+noccA, e+noccA);
+                 FabA->set(a, e, value);
+             }
+        }
+
+        // F_ae -=  \sum_{Q,m} Tau'_ma^Q b_me^Q
+        FabA->contract(true, false, navirA, navirA, nQ * naoccA, Tau, bQiaA, -1.0, 1.0);
+        Tau.reset();
+    }// if (reference_ == "RESTRICTED")
+
+    // UHF
+    else if (reference_ == "UNRESTRICTED") {
+        SharedTensor2d Tau, T;
+
+        // F(M,I) OO Alpha Block
+        // F(M,I) = (1-kronDelta(M,I)) f(M,I) + \sum_(E) 0.5*t(I,E)*f(M,E) + \sum_(Q) tQ*b(Q,MI) + \sum_(Q,E) Tau"(Q,IE)*b(Q,ME)
```

Comment:
Make this file consistent on `delta` vs `kronDelta`, and on whether the _overall_ F is defined before each piece is. This is inconsistent between the restricted and unrestricted parts, at present.